HBFA/UefiHostFuzzTestCasePkg/TestCase/MdeModulePkg/Universal/CapsulePei/Common/TestCapsulePei.c

```diff
@@ -67,7 +67,7 @@ RunTestHarness(
   MemoryBase = AllocatePool (MemorySize);
 
   MemoryResource[0].PhysicalStart  = (UINT64)(UINTN)TestBuffer;
-  MemoryResource[0].ResourceLength = (UINT64)(UINTN)TOTAL_SIZE;
+  MemoryResource[0].ResourceLength = (UINT64)(UINTN)TestBufferSize;

```

Comment:
Don't you also need to check that TestBufferSize<TOTAL_SIZE ?

Reply:
No, it's edk2's job to validate that accesses stay within the resource boundaries (which evidently it wasn't doing well). However, if you set TOTAL_SIZE here and libfuzzer generates a buffer less then that can still trigger an invalid memory access.